docs/source/newsfragments/4543.change.rst

```diff
@@ -0,0 +1 @@
+Settings values too large for :class:`~cocotb.handle.IntegerObject`, :class:`~cocotb.handle.EnumObject`, and :class:`~cocotb.handle.LogicArrayObject` now raise :exc:`ValueError` instead of :exc:`OverflowError`.
```

Comment:
```suggestion
Setting values too large for :class:`~cocotb.handle.IntegerObject`, :class:`~cocotb.handle.EnumObject`, and :class:`~cocotb.handle.LogicArrayObject` now raise :exc:`ValueError` instead of :exc:`OverflowError`.
```
?